torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2974,10 +2978,10 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                     ), "Currently the way for _is_sharded to be False is \
                         world_size == 1 or sharding_stratagy is set to be NO_SHARD"
                     if self.sharding_strategy == ShardingStrategy.NO_SHARD:
-                        dist.all_reduce(param.grad, group=self.process_group)
-                        if self.gradient_postdivide_factor > 1:
-                            # Average grad by world_size for consistency with PyTorch DDP.
-                            param.grad.div_(self.gradient_postdivide_factor)
+                        # if a communication hook was not registered,
+                        # then a default hook (`all_reduce`) will be used
+                        self.communication_hook(self.communication_hook_state, param.grad)
```

Comment:
nit: in your next PR, would you please assert 'self.communication_hook' and 'self.communication_hook_state' is not None as well before calling them?

Reply:
sure, thank you for pointing that out